bytecomp/matching.ml

```diff
@@ -1295,6 +1344,47 @@ and dont_precompile_var args cls def k =
     k )
 
 and precompile_or argo cls ors args def k =
+  (* Let's say the following pm was given to [split_and_precompile]
+     {[
+       match e1           ,  e2 ,  e3 with
+       | C1(arg11, arg21) ,  p1 ,  q1 -> lam1
+       | C2(arg12)        ,  p2 ,  q2 -> lam2
+       | ( C1(_, argO)
+         | C2(argO)   )   ,  p3 ,  q3 -> lam3
+       | C3               ,  p4 ,  q4 -> lam4
+       | C1(_, _)         ,  _  ,  _  -> lam5
```

Comment:
The `_, _` part before `lam5` creates a risk of confusing some of the `_, _` which may be copies of this one or have been inserted by the compilation technique itself. What about using `p5, q5` here? (Were you just fed up with the labeling or is there a reason to have catch-alls here?)

Reply:
Actually, it's meant to indicate wildcards in the source, and the point of having wildcard in the source of the example is: that way we're sure the row won't be moved above the or-matrix.
I'd be happier with `p5, q5`, but we would have to add a note saying `p5 <= p4 && q5 <= q4` .